### PR TITLE
move webpack/karma configuration to config/ folder

### PR DESCRIPTION
**Pull request by garbas**

### CODE REVIEW

File: config/webpack.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+module.exports = rootpath => {
+
+  const SRC = path.resolve(rootpath, 'src');
+  const BUILD = path.resolve(rootpath, 'build');
+  const IS_PROD = process.env.npm_lifecycle_event === 'build';
+
+  let plugins = [chunk, define];
+
+  if (IS_PROD) {
+    plugins.push(...[clean, uglify])
```
Comment (eliperelman) [inline config/webpack.js]: I don't think Node v4 supports array spread, so this will need to be separate `push`es.

File: config/webpack.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+const css = require('./css');
+const define = require('./define');
+const html = require('./html');
+const js = require('./js');
+const lint = require('./lint');
+const path = require('path');
+const uglify = require('./uglify');
+
+module.exports = rootpath => {
+
```
Comment (eliperelman) [inline config/webpack.js]: nit: remove this extra line.

File: config/karma.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+const merge = require('webpack-merge');
+const js = require('./js');
+const json =  require('./json');
+const package_ = require('./package');
```
Comment (eliperelman) [inline config/karma.js]: Let's rename this to `pkg` to avoid the dangling underscore.

File: config/js.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
 module.exports = {
   test: /\.jsx?$/,
   exclude: /(node_modules|bower_components|build)/,
-  loaders: ['babel']
+  loaders: ['babel?presets[]=es2015&presets[]=stage-0&presets[]=react']
```
Comment (eliperelman) [inline config/js.js]: I prefer having this in a .babelrc, but if that won't work then I'm OK with leaving it in here. If we define it in-file like this, let's switch the syntax to:

```js
loader: 'babel',
query: {
  presets: [
    'es2015',
    'stage-0',
    'react'
  ]
}
```

File: config/karma.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+const istanbul =  require('./istanbul');
+
+module.exports = (webpackConfig, rootpath) => {
+  delete webpackConfig.plugins;
+  return {
```
Comment (eliperelman) [inline config/karma.js]: In my testing, I don't think we need `rootpath` or `basePath` at all.

File: config/js.js [hunk 6]
```diff
@@ -1,5 +1,5 @@
 module.exports = {
   test: /\.jsx?$/,
   exclude: /(node_modules|bower_components|build)/,
```
Comment (eliperelman) [inline config/js.js]: So I figured out why Travis bombs on running the tests. Could you change this `exclude` to remove the `build` directory?

```js
exclude: /(node_modules/bower_components)/
```

File: config/karma.js (hunk 5 above)
Comment (eliperelman) [inline config/karma.js]: Insert newline between `delete` and `return`.

Comment (garbas) [inline config/karma.js]: ``rootpath`` would be needed if you want to have webpack.config.js not at the root of the repository

**Comment by garbas** [comment]
@eliperelman i've fixed requested changes and squashed everything into one commit

**Comment by garbas** [comment]
@eliperelman i've also added new commit with changes we were talking (merging webpack into one file)

**Comment by eliperelman** [comment]
I'll go ahead and merge then push a follow-up commit for the Travis fixes.

**Pull request merged**